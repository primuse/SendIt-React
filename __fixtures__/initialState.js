export const props = {
  auth: {
    isLoading: false,
    isAuthenticated: true,
    user: {
      firstName: 'Tiku',
      lastName: 'Okoye',
      registered: '22-03-2019',
      email: 'cim@gmail.com',
      isadmin: false
    }
  },
  parcel: {
    parcels: [
      {
        id: '1',
        parcelName: 'rice',
        weigth: '30',
        price: '4,000',
        destination: 'Owerri',
        receiver: 'Tiku Okoye',
        senton: '22-03-2019',
        status: 'created'
      },
      {
        id: '1',
        parcelName: 'rice',
        weigth: '30',
        price: '4,000',
        destination: 'Owerri',
        receiver: 'Tiku Okoye',
        senton: '22-03-2019',
        status: 'created'
      }
    ],
    createdParcels: [],
    deliveredParcels: [],
    inTransitParcels: [],
    canceledParcels: [],
    isLoading: false,
    userParcel: {
      id: 2,
      parcelName: 'rice',
      weigth: '30',
      price: '4,000',
      destination: 'Owerri',
      receiver: 'Tiku Okoye',
      senton: '22-03-2019',
      status: 'created'
    }
  },
  users: {
    allUsers: [
      {
        id: '1',
        firstName: 'Tiku',
        lastName: 'Okoye',
        isadmin: 'false',
        email: 'cim@gmail.com',
        registered: '22-03-2019',
      },
      {
        id: '1',
        firstName: 'Tiku',
        lastName: 'Okoye',
        isadmin: 'false',
        email: 'cim@gmail.com',
        registered: '22-03-2019',
      }
    ],
    show: false
  },
  user: {
    firstName: 'Tiku',
    lastName: 'Okoye',
    registered: '22-03-2019',
    email: 'cim@gmail.com',
    isadmin: false
  }
};

export const props2 = {
  auth: {
    isLoading: false,
    isAuthenticated: true,
    user: {
      firstName: 'Tiku',
      lastName: 'Okoye',
      registered: '22-03-2019',
      email: 'cim@gmail.com',
      isadmin: true
    }
  },
  parcel: {
    parcels: [
      {
        id: '1',
        parcelName: 'rice',
        weigth: '30',
        price: '4,000',
        destination: 'Owerri',
        receiver: 'Tiku Okoye',
        senton: '22-03-2019',
        status: 'created'
      },
      {
        id: '1',
        parcelName: 'rice',
        weigth: '30',
        price: '4,000',
        destination: 'Owerri',
        receiver: 'Tiku Okoye',
        senton: '22-03-2019',
        status: 'created'
      }
    ],
    createdParcels: [],
    deliveredParcels: [],
    inTransitParcels: [],
    canceledParcels: [],
    isLoading: false,
    userParcel: {
      id: 2,
      parcelName: 'rice',
      weigth: '30',
      price: '4,000',
      destination: 'Owerri',
      receiver: 'Tiku Okoye',
      senton: '22-03-2019',
      status: 'created'
    }
  },
  users: {
    allUsers: [
      {
        id: '1',
        firstName: 'Tiku',
        lastName: 'Okoye',
        isadmin: 'false',
        email: 'cim@gmail.com',
        registered: '22-03-2019',
      },
      {
        id: '1',
        firstName: 'Tiku',
        lastName: 'Okoye',
        isadmin: 'false',
        email: 'cim@gmail.com',
        registered: '22-03-2019',
      }
    ],
    show: false
  },
  user: {
    firstName: 'Tiku',
    lastName: 'Okoye',
    registered: '22-03-2019',
    email: 'cim@gmail.com',
    isadmin: true
  }
};

export const props3 = {
  parcel: {
    isLoading: true
  }
};
