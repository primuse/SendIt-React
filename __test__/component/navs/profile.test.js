import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Profile from '../../../src/components/navs/profile.jsx';
import { props } from '../../../__fixtures__/initialState';

describe('Profile Page component', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Profile user={props.user} parcel={props.parcel} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
