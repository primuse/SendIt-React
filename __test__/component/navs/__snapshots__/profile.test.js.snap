// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Page component renders correctly 1`] = `
<div
  className="d-flex"
  id="user-cont"
>
  <div
    id="user-desc"
  >
    <img
      id="user-img"
      src="test-file-stub"
    />
    <div
      className="text-center"
    >
      <h2
        id="id"
      >
        SDO
      </h2>
      <span
        className="d-block mb-14"
        id="profile-name"
      >
        <h4
          className="d-inline"
          id="firstName"
        />
        <h4
          className="d-inline"
          id="lastName"
        />
      </span>
      <small
        id="email"
      >
        cim@gmail.com
      </small>
      <h5
        id="registered"
      >
        Registered On 
        22-03-2019
      </h5>
    </div>
  </div>
  <div
    className="text-center"
    id="user-info"
  >
    <h1
      className="mb-6"
    >
      My Parcels
    </h1>
    <div
      className="d-flex content-between mb-6"
    >
      <div>
        <small>
          Created
        </small>
        <h4
          id="created"
        >
          0
        </h4>
      </div>
      <div>
        <small>
          Delivered
        </small>
        <h4
          className="green"
          id="delivered"
        >
          0
        </h4>
      </div>
      <div>
        <small>
          In-transit
        </small>
        <h4
          className="blue"
          id="in-transit"
        >
          0
        </h4>
      </div>
      <div>
        <small>
          Cancelled
        </small>
        <h4
          className="red"
          id="canceled"
        >
          0
        </h4>
      </div>
    </div>
  </div>
</div>
`;
